engine/crates/runtime-local/src/rate_limiting/redis.rs

```diff
@@ -167,7 +158,7 @@ impl RedisRateLimiter {
     async fn limit_inner(&self, context: &dyn RateLimiterContext) -> Result<(), Error> {
         let Some(key) = context.key() else { return Ok(()) };
 
-        let Some(config) = self.subgraph_limits.get(key) else {
+        let Some(config) = self.limits.borrow().get(key).copied() else {
```

Comment:
:zap: **Suggestion** 
Consider handling the case where `self.limits.borrow()` might fail, to avoid potential panics.